### PR TITLE
build(docker): build & publish GitHub Package

### LINKED ISSUE #1548: Move Docker containers under /linkedin or /datahub account
Right now, all Docker images are deployed to my personal Docker Hub account. This is not quite maintainable and doesn't look professional as well. At the time, I wasn't able to use Linkedin's organization account (owner left the company) to deploy images, therefore, I've deployed under my personal repo.

@mars-lan is now one of the owner of LinkedIn organization account in Docker Hub. We should consider starting to deploy Docker images to there. Alternatively, we can create a new organization called DataHub and deploy images there.

### PR DESCRIPTION
**Pull request by mars-lan**
Will keep publishing to Docker Hub meanwhile until all Dockerfiles have been updated to point to GitHub.
Fixes #1548

Check out both the sample output: https://github.com/mars-lan/datahub/packages

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by jplaisted** [comment]
Any particular reason to use github rather than docker hub? This just becomes an extra step for `docker pull`, afaik (gotta specify github now as a repo)

**Comment by jplaisted** [comment]
Or can we just change the docker-compose file `image:` tag?

**Comment by mars-lan** [comment]
> Or can we just change the docker-compose file `image:` tag?

yup

**Comment by mars-lan** [comment]
> Any particular reason to use github rather than docker hub? This just becomes an extra step for `docker pull`, afaik (gotta specify github now as a repo)

There's a few benefits of hosting everything on GitHub, including better analytics and linking other projects that uses the docker images to us etc.

File: .github/workflows/docker-mce-consumer.yml [hunk 1]
```diff
@@ -12,10 +12,21 @@ jobs:
...
           dockerfile: ./docker/mce-consumer/Dockerfile
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
           repository: linkedin/datahub-mce-consumer
-          tag_with_ref: true
+          tags: ${{ steps.tag.outputs.tag }}
+      - uses: VaultVulp/gp-docker-action@1.1.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          dockerfile: ./docker/gms/Dockerfile
```
Comment (keremsahin1) [inline .github/workflows/docker-mce-consumer.yml:30]: This seems wrong.

File: .github/workflows/docker-mae-consumer.yml [hunk 2]
```diff
@@ -12,10 +12,21 @@ jobs:
...
           dockerfile: ./docker/mae-consumer/Dockerfile
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
           repository: linkedin/datahub-mae-consumer
-          tag_with_ref: true
+          tags: ${{ steps.tag.outputs.tag }}
+      - uses: VaultVulp/gp-docker-action@1.1.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          dockerfile: ./docker/gms/Dockerfile
```
Comment (keremsahin1) [inline .github/workflows/docker-mae-consumer.yml:30]: This seems wrong.

File: .github/workflows/docker-frontend.yml [hunk 3]
```diff
@@ -12,10 +12,21 @@ jobs:
...
           dockerfile: ./docker/frontend/Dockerfile
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
           repository: linkedin/datahub-frontend
-          tag_with_ref: true
+          tags: ${{ steps.tag.outputs.tag }}
+      - uses: VaultVulp/gp-docker-action@1.1.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          dockerfile: ./docker/gms/Dockerfile
```
Comment (keremsahin1) [inline .github/workflows/docker-frontend.yml:30]: This seems wrong.

File: .github/workflows/docker-mce-consumer.yml (hunk 1 above)
Comment (jplaisted) [reply .github/workflows/docker-mce-consumer.yml:30]: Can you more details to these comments? Why is it wrong?

Comment (mars-lan) [reply .github/workflows/docker-mce-consumer.yml:30]: Wrong dockerfile. Will fix that.

**Pull request merged**